hazelcast/src/main/java/com/hazelcast/internal/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -1496,6 +1497,20 @@ private void processNewState(Collection<MigrationInfo> allCompletedMigrations,
                     applyNewPartitionTable(latestPartitions, allCompletedMigrations, thisAddress);
                 }
                 shouldFetchPartitionTables = false;
+            } catch (Throwable rethrowed) {
+                String lineSeparator = System.lineSeparator();
+
+                StringBuilder sb = new StringBuilder()
+                        .append("latestPartitions:").append(lineSeparator)
+                        .append(StringUtil.toString(latestPartitions)).append(lineSeparator)
+                        .append("allCompletedMigrations:").append(lineSeparator)
+                        .append(StringUtil.toString(allCompletedMigrations)).append(lineSeparator)
+                        .append("allActiveMigrations:").append(lineSeparator)
+                        .append(StringUtil.toString((allActiveMigrations))).append(lineSeparator)
+                        .append(rethrowed);
+
+                logger.warning(sb.toString());
```

Comment:
This will cause duplicate logging(since we already log exceptions) for the same exception although this one has a detailed message. Maybe we can print this as a debug level and set log level of test as debug.

Reply:
thanks, makes sense. But doesn't setting log level to debug print almost everything and remove potential concurrency issues? And since exceptions on this part of the code occur very rarely, I think we can afford printing it as warning. wdyt?